Source/scrollrt.cpp

```diff
@@ -1663,233 +1663,203 @@ void __fastcall scrollrt_draw_upper(int x, int y, int sx, int sy, int a5, int a6
-	int v15;            // ecx
-	int v16;            // edx
-	char v17;           // bl
-	ItemStruct *v18;    // esi
-	char *v19;          // eax
-	signed int v20;     // ebx
-	int v21;            // ebx
-	unsigned int v22;   // ecx
-	PlayerStruct *v23;  // esi
-	int v24;            // esi
-	int v25;            // eax
-	MonsterStruct *v26; // esi
-	CMonster *v27;      // ecx
-	int v28;            // ebx
-	int v29;            // edi
-	unsigned int v30;   // ecx
-	PlayerStruct *v31;  // esi
-	int v32;            // esi
-	int v33;            // eax
-	MonsterStruct *v34; // esi
-	CMonster *v35;      // ecx
-	int v36;            // ebx
-	int v37;            // edi
-	ItemStruct *v38;    // esi
-	char *v39;          // ecx
-	signed int v40;     // ebx
-	int v41;            // ebx
-	int v42;            // [esp+Ch] [ebp-18h]
-	int v43;            // [esp+10h] [ebp-14h]
-	char *dst_buf;      // [esp+14h] [ebp-10h]
-	int xa;             // [esp+18h] [ebp-Ch]
-	char v46;           // [esp+1Dh] [ebp-7h]
-	char v47;           // [esp+1Eh] [ebp-6h]
-	char v48;           // [esp+1Fh] [ebp-5h]
-	char v49;           // [esp+20h] [ebp-4h]
-	char v50;           // [esp+21h] [ebp-3h]
-	char v51;           // [esp+22h] [ebp-2h]
-	char v52;           // [esp+23h] [ebp-1h]
-
-	xa = x;
-	dst_buf = buffer;
-	v8 = 112 * x + y;
-	v9 = dDead[0][v8];
-	v52 = dFlags[0][v8];
-	v49 = dObject[0][v8];
-	v51 = dItem[0][v8];
-	v10 = dPlayer[0][v8 - 1];
-	v50 = dPlayer[0][v8];
-	v48 = dArch[0][v8];
-	v11 = dung_map[0][v8];
-	v12 = (int *)((char *)dMonster + 4 * v8);
-	v46 = v11;
-	v47 = v10;
-	v42 = *v12;
-	v43 = *(v12 - 1);
-	if (visiondebug && v52 & DFLAG_LIT)
-		CelDecodeHdrOnly((BYTE *)dst_buf, (BYTE *)pSquareCel, 1, 64, 0, a5);
-	if (MissilePreFlag && v52 & DFLAG_MISSILE)
-		DrawMissile(xa, y, sx, sy, 0, a5, TRUE);
-	if (light_table_index < lightmax) {
-		if (v9) {
-			v13 = &dead[(v9 & 0x1F) - 1];
-			v14 = (int *)v13->_deadData[(v9 >> 5) & 7];
-			v15 = sx - v13->_deadWidth2;
-			if (v14) {
-				v16 = v13->_deadFrame;
-				if (v16 >= 1 && (unsigned int)*v14 <= 0x32 && v16 <= *v14) {
-					v17 = v13->_deadtrans;
-					if (v17)
-						Cl2DecodeFrm3(v15, sy, (BYTE *)v14, v16, v13->_deadWidth, 0, a5, v17);
-					else
-						Cl2DecodeLightTbl(v15, sy, (BYTE *)v14, v16, v13->_deadWidth, 0, a5);
+	int px, py, nCel, nMon, negMon, p;
+	char bFlag, bDead, bObj, bItem, bPlr, bArch, bMap, negPlr, dd;
+	DeadStruct *pDeadGuy;
+	ItemStruct *pItem;
+	PlayerStruct *pPlayer;
+	MonsterStruct *pMonster;
+	BYTE *pCelBuff;
+	DWORD *pFrameTable;
+
+	/// ASSERT: assert((DWORD)sx < MAXDUNX);
+	/// ASSERT: assert((DWORD)sy < MAXDUNY);
+	bFlag = dFlags[sx][sy];
+	bDead = dDead[sx][sy];
+	bObj = dObject[sx][sy];
+	bItem = dItem[sx][sy];
+	bPlr = dPlayer[sx][sy];
+	bArch = dArch[sx][sy];
+	bMap = dung_map[sx][sy];
+	nMon = dMonster[sx][sy];
+
+	/// ASSERT: assert((DWORD)(sy-1) < MAXDUNY);
+	negPlr = dPlayer[sx][sy - 1];
+	negMon = dMonster[sx][sy - 1];
+
+	if(visiondebug && (bFlag & 0x40)) {

```

Comment:
Enums :)

```c
typedef enum dflag {
	DFLAG_MISSILE     = 0x01,
	DFLAG_VISIBLE     = 0x02,
	DFLAG_DEAD_PLAYER = 0x04,
	DFLAG_POPULATED   = 0x08,
	DFLAG_MONSTER     = 0x10,
	DFLAG_PLAYER      = 0x20,
	DFLAG_LIT         = 0x40,
	DFLAG_EXPLORED    = 0x80,
} dflag;
```

Reply:
I don't have the branch anymore so I'll add those later :)